File: DS/Linkedlist/singly_linked_list/C/delete_node.c
#include <stdio.h>
#include <stdlib.h>

struct Node{
    int data;
    struct Node *link;
};

struct Node *head; //pointer to head node

// insert node at the end of linked list
void insert_end(int num){
    struct Node *temp = (struct Node*)malloc(sizeof(struct Node));
    temp->data = num;
    if(head == NULL){
        temp->link = NULL;
        head = temp;
    }

    else{
        temp->link = NULL;
        struct Node *temp1 = head;
        while (temp1->link!=NULL)
        {
            temp1 = temp1->link;
        }

        temp1->link = temp;  
    } 
}

// delete function
void delete(int loc){
    struct Node *temp1 = (struct Node*)malloc(sizeof(struct Node));
    temp1 = head;

    if(loc == 1){
       head = temp1->link;
       free(temp1);
    }

    else{
        loc -= 2;

        while (loc--)
        {
           temp1 = temp1->link; 
        }
        struct Node *temp = temp1->link;
        temp1->link = temp->link;
        free(temp);
    }
    
}

// print list
void print_list(){
    struct Node *temp = head;
     while (temp!=NULL)
     {
        printf("%d ", temp->data);
        temp = temp->link;
     }
}


int main(){
    int size;
    scanf("%d", &size);

    while (size--)
    {
        int num;
        scanf("%d", &num);
        insert_end(num);
    }

    delete(3);
    
    print_list();
}

File: DS/Linkedlist/singly_linked_list/PYTHON/delete_node.py
class Node:
    def __init__(self, num):
        self.data = num
        self.link = None


class LinkedList:

    def __init__(self):
        self.head = None

    # insert node at the end of linked list
    def append(self, num):
        temp = Node(num)
        if self.head is None:
            self.head = temp
            return

        temp1 = self.head
        while temp1.link is not None:
            temp1 = temp1.link

        temp1.link = temp

    # delete node
    def delete_node(self, loc):
        temp = self.head
        if loc == 1:
            self.head = temp.link
            temp = None
            return
        loc -= 2
        while loc:
            temp = temp.link
            loc -= 1
        temp1 = temp.link
        temp.link = temp1.link
        temp1 = None

    # print
    def print_list(self):
        temp = self.head
        while temp:
            print(temp.data, end=" ")
            temp = temp.link
          
if __name__ == '__main__':

    l_list = LinkedList()

    l_list.append(1)
    l_list.append(2)
    l_list.append(3)
    l_list.append(4)
    l_list.append(5)
    
    l_list.delete_node(1)
    
    l_list.print_list()
